client/dashboard/sites/site/environment-switcher.tsx

```diff
@@ -0,0 +1,195 @@
+import { useQuery, useMutation } from '@tanstack/react-query';
+import {
+	__experimentalHStack as HStack,
+	Button,
+	Dropdown,
+	MenuGroup,
+	MenuItem,
+	Spinner,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { sprintf, __ } from '@wordpress/i18n';
+import { Icon, chevronDownSmall, plus } from '@wordpress/icons';
+import { store as noticesStore } from '@wordpress/notices';
+import { SVG, Path } from '@wordpress/primitives';
+import { siteByIdQuery } from '../../app/queries/site';
+import { stagingSiteCreateMutation } from '../../app/queries/site-staging-sites';
+import RouterLinkMenuItem from '../../components/router-link-menu-item';
+import { canManageSite, canCreateStagingSite } from '../features';
+import type { Site } from '../../data/types';
+
+type Env = 'production' | 'staging';
+
+const production = (
```

Comment:
WDYT about moving these shapes to component level since they're used elsewhere in the Hosting Dashboard, like QOBjujZah0UlGDDdLKZhzi-fi-1059_63142 or QOBjujZah0UlGDDdLKZhzi-fi-2587_181484 or QOBjujZah0UlGDDdLKZhzi-fi-6785_228323